pulsar-broker/src/main/java/com/yahoo/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -150,9 +150,14 @@ private NamespaceBundle getFullBundle(NamespaceName fqnn) throws Exception {
         return bundleFactory.getFullBundle(fqnn);
     }
 
-    private static final Deserializer<ServiceLookupData> serviceLookupDataDeserializer = (key, content) ->
-            jsonMapper().readValue(content, ServiceLookupData.class);
-
+    private static final Deserializer<ServiceLookupData> serviceLookupDataDeserializer = (key, content) -> {
+        final String jsonString = new String(content);
+        if (jsonString.contains("\"allocatedCPU\":")) {
```

Comment:
Yes, I would prefer that. Check which implementation is active and then try both format, starting from the active one and falling back to the inactive.